source/priority-queue.ts

```diff
@@ -32,6 +32,27 @@ export default class PriorityQueue implements Queue<RunFunction, PriorityQueueOp
 		this.#queue.splice(index, 0, element);
 	}
 
+	setPriority(id: string, priority?: number) {
+		const existingIndex: number = this.#queue.findIndex((element: Readonly<PriorityQueueOptions>) => element.id === id);
+		const [item] = this.#queue.splice(existingIndex, 1);
+		if (item === undefined) {
+			return;
+		}
+
+		item.priority = priority ?? ((item.priority ?? 0) + 1);
+		if (this.size && this.#queue[this.size - 1]!.priority! >= priority!) {
```

Comment:
It looks like the check for `this.size` ensures that it exists. However, it's already a required property:

https://github.com/sindresorhus/p-queue/blob/47a4f19ab3d4422e06cb1b618f4c45a11b5def30/source/queue.ts#L4

So, if the check fails, we have bigger problems and might even want to throw an error.

This shortcut would also be incorrectly passed by if `this.size === 0`.

```suggestion
		if (this.size === 0 || this.#queue[this.size - 1]!.priority! >= priority!) {
```

Reply:
sounds right! shall we update this at #L23 too, since this change is taken from there, or should we create a common function to position the promise function at right place in queue wrt priority?
Please do suggest a better name for this function in case we want to move with latter. :)

Reply:
> should we create a common function to position the promise function at right place in queue wrt priority?
Let create a private `_enqueueElement` method.

Reply:
reused `enqueue` method itself as there was no additional code in the same.